codegen-server-test/build.gradle.kts

```diff
@@ -0,0 +1,117 @@
+    implementation("software.amazon.smithy:smithy-aws-traits:$smithyVersion")
+}
+
+data class CodegenTest(val service: String, val module: String, val extraConfig: String? = null)
+
+val CodegenTests = listOf(CodegenTest("com.amazonaws.simple#SimpleService", "simple"))
+
+fun generateSmithyBuild(tests: List<CodegenTest>): String {
+    val projections =
+            tests.joinToString(",\n") {
+                """
+            "${it.module}": {
+                "plugins": {
+                    "rust-server-codegen": {
+                      "codegen": {
+                        "includeFluentClient": false
+                      },
+                      "runtimeConfig": {
+                        "relativePath": "${rootProject.projectDir.absolutePath}/rust-runtime"
+                      },
+                      "service": "${it.service}",
+                      "module": "${it.module}",
+                      "moduleVersion": "0.0.1",
+                      "moduleAuthors": ["protocoltest@example.com"]
+                      ${it.extraConfig ?: ""}
+                 }
+               }
+            }
+        """.trimIndent()
+            }
+    return """
+    {
+        "version": "1.0",
+        "projections": { $projections }
+    }
+    """
+}
+
+task("generateSmithyBuild") {
+    description = "generate smithy-build.json"
+    doFirst { projectDir.resolve("smithy-build.json").writeText(generateSmithyBuild(CodegenTests)) }
+}
+
+fun generateCargoWorkspace(tests: List<CodegenTest>): String {
+    return """
+    [workspace]
+    members = [
+        ${tests.joinToString(",") { "\"${it.module}/rust-server-codegen\"" }}
+    ]
+    """.trimIndent()
+}
+
+task("generateCargoWorkspace") {
+    description = "generate Cargo.toml workspace file"
+    doFirst {
+        buildDir.resolve("smithyprojections/codegen-server-test/Cargo.toml")
+                .writeText(generateCargoWorkspace(CodegenTests))
+    }
+}
+
+tasks["smithyBuildJar"].dependsOn("generateSmithyBuild")
+
+tasks["assemble"].finalizedBy("generateCargoWorkspace")
+
+tasks.register<Exec>("cargoCheck") {
+    workingDir("build/smithyprojections/codegen-server-test/")
+    // disallow warnings
+    environment("RUSTFLAGS", "-D warnings")
+    commandLine("cargo", "check")
+    dependsOn("assemble")
+}
+
+tasks.register<Exec>("cargoTest") {
+    workingDir("build/smithyprojections/codegen-server-test/")
+    // disallow warnings
+    environment("RUSTFLAGS", "-D warnings")
+    commandLine("cargo", "test")
+    dependsOn("assemble")
+}
+
+tasks.register<Exec>("cargoDocs") {
+    workingDir("build/smithyprojections/codegen-server-test/")
+    // disallow warnings
+    environment("RUSTFLAGS", "-D warnings")
+    commandLine("cargo", "doc", "--no-deps")
+    dependsOn("assemble")
+}
+
+tasks.register<Exec>("cargoClippy") {
+    workingDir("build/smithyprojections/codegen-server-test/")
+    // disallow warnings
+    commandLine("cargo", "clippy")
```

Comment:
@rcoh I've noticed that in the `codegen-test` project's `build.gradle.kts`, [we error out on clippy warnings](https://github.com/awslabs/smithy-rs/blob/f9771d517bb5978c33ea0c738f4fcdd178d16823/codegen-test/build.gradle.kts#L156), and here we don't.

Yet as of writing, if I run `./gradlew :codegen-test:build`, the build fails because of Clippy warnings. Why, and how come CI works despite this?

Reply:
I have changed the clippy config for this project since it was erroring out as you noticed. I did not dig into why this is happening yet.